src/hotspot/share/gc/parallel/parallelScavengeHeap.hpp

```diff
@@ -286,4 +290,19 @@ class AdaptiveSizePolicyOutput : AllStatic {
   }
 };
 
+// The HeapBlockClaimer is used during parallel iteration over heap,
+// allowing workers to claim heap blocks, gaining exclusive rights to these blocks.
+// The eden and survivor spaces are treated as single blocks as it is hard to divide
+// these spaces.
+// The old spaces are divided into serveral fixed-size blocks.
+class HeapBlockClaimer : public StackObj {
+  int _claimed_index;
```

Comment:
There is severe type confusion throughout the code about this. Sometimes the code implicitly casts to uint, sometimes relying on automatic widening to do the right thing.

Also, please use a larger type on 64 bits as the range of int plus the block size can not span the whole 64 bits, only 52 bits. While "large", there seems no reason to simply use something like `ssize_t` instead (if signedness is required).